`Send diagnostic` report does not attach the device logs in Google Pixel 7a

**Describe the bug**
* The `Send diagnostic` report does not attach the device logs in Google Pixel 7a. However, it is working fine on my other Samsung device.
* Also when there is a large logs available then application failed to send the diagnostic report on email.
```kotlin
!!! FAILED BINDER TRANSACTION !!!  (parcel size = 13932232)
2024-11-20 16:23:04.817 28305-28305 AndroidRuntime          org.kiwix.kiwixmobile                E  FATAL EXCEPTION: main
                                                                                                    Process: org.kiwix.kiwixmobile, PID: 28305
                                                                                                    java.lang.RuntimeException: android.os.TransactionTooLargeException: data parcel size 13932232 bytes
                                                                                                    	at android.app.ApplicationPackageManager.queryIntentActivitiesAsUser(ApplicationPackageManager.java:1511)
                                                                                                    	at android.app.ApplicationPackageManager.queryIntentActivities(ApplicationPackageManager.java:1486)
                                                                                                    	at org.kiwix.kiwixmobile.core.compat.CompatV33.queryIntentActivities(CompatV33.kt:38)
                                                                                                    	at org.kiwix.kiwixmobile.core.compat.CompatHelper$Companion.queryIntentActivitiesCompat(CompatHelper.kt:63)
                                                                                                    	at org.kiwix.kiwixmobile.core.error.ErrorActivity$setupReportButton$1$1.invokeSuspend(ErrorActivity.kt:100)
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:959)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:100)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:232)
                                                                                                    	at android.os.Looper.loop(Looper.java:317)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8705)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:886)
                                                                                                    	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@3638d6a, Dispatchers.Main.immediate]
                                                                                                    Caused by: android.os.TransactionTooLargeException: data parcel size 13932232 bytes
                                                                                                    	at android.os.BinderProxy.transactNative(Native Method)
                                                                                                    	at android.os.BinderProxy.transact(BinderProxy.java:586)
                                                                                                    	at android.content.pm.IPackageManager$Stub$Proxy.queryIntentActivities(IPackageManager.java:5203)
                                                                                                    	at android.app.ApplicationPackageManager.queryIntentActivitiesAsUser(ApplicationPackageManager.java:1501)
   ```

https://github.com/user-attachments/assets/997b692f-3cf7-4a90-b153-a3d557df119b


**Expected behavior**
It should attach the device logs in the report.

**Environment**
- Version of Kiwix Android: 3.11.1
- Device: Google Pixel 7a
- OS version: Android 14

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/error/ErrorActivity.kt
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/error/ErrorActivity.kt
@@ -19,12 +19,13 @@ package org.kiwix.kiwixmobile.core.error
 
 import android.annotation.SuppressLint
 import android.content.Intent
-import android.net.Uri
+import android.content.pm.ResolveInfo
 import android.os.Build
 import android.os.Bundle
 import android.os.Process
 import androidx.activity.result.contract.ActivityResultContracts
 import androidx.core.content.ContextCompat
+import androidx.core.content.FileProvider
 import androidx.lifecycle.lifecycleScope
 import kotlinx.coroutines.launch
 import org.kiwix.kiwixmobile.core.CoreApp.Companion.coreComponent
@@ -96,16 +97,66 @@ open class ErrorActivity : BaseActivity() {
     activityKiwixErrorBinding?.reportButton?.setOnClickListener {
       lifecycleScope.launch {
         val emailIntent = emailIntent()
-        val activities =
-          packageManager.queryIntentActivitiesCompat(emailIntent, ResolveInfoFlagsCompat.EMPTY)
-        if (activities.isNotEmpty()) {
-          sendEmailLauncher.launch(Intent.createChooser(emailIntent, "Send email..."))
+        val activities = getSupportedEmailApps(emailIntent, supportedEmailPackages)
+        val targetedIntents = createEmailIntents(emailIntent, activities)
+        if (activities.isNotEmpty() && targetedIntents.isNotEmpty()) {
+          val chooserIntent =
+            Intent.createChooser(targetedIntents.removeFirst(), "Send email...")
+          chooserIntent.putExtra(Intent.EXTRA_INITIAL_INTENTS, targetedIntents.toTypedArray())
+          sendEmailLauncher.launch(chooserIntent)
         } else {
           toast(getString(R.string.no_email_application_installed))
         }
       }
     }
   }
+
+  /**
+   * Get a list of supported email apps.
+   */
+  private fun getSupportedEmailApps(
+    emailIntent: Intent,
+    supportedPackages: List<String>
+  ): List<ResolveInfo> {
+    return packageManager.queryIntentActivitiesCompat(emailIntent, ResolveInfoFlagsCompat.EMPTY)
+      .filter {
+        supportedPackages.any(it.activityInfo.packageName::contains)
+      }
+  }
+
+  /**
+   * Create a list of intents for supported email apps.
+   */
+  private fun createEmailIntents(
+    emailIntent: Intent,
+    activities: List<ResolveInfo>
+  ): MutableList<Intent> {
+    return activities.map { resolveInfo ->
+      Intent(emailIntent).apply {
+        setPackage(resolveInfo.activityInfo.packageName)
+      }
+    }.toMutableList()
+  }
+
+  // List of supported email apps
+  private val supportedEmailPackages = listOf(
+    "com.google.android.gm",               // Gmail
+    "com.zoho.mail",                       // Zoho Mail
+    "com.microsoft.office.outlook",        // Outlook
+    "com.yahoo.mobile.client.android.mail",// Yahoo Mail
+    "me.bluemail.mail",                    // BlueMail
+    "ch.protonmail.android",               // ProtonMail
+    "com.fsck.k9",                         // K-9 Mail
+    "com.maildroid",                       // Maildroid
+    "org.kman.AquaMail",                   // Aqua Mail
+    "com.edison.android.mail",              // Edison Mail
+    "com.readdle.spark",                   // Spark
+    "com.gmx.mobile.android.mail",          // GMX Mail
+    "com.fastmail",                        // FastMail
+    "ru.mail.mailapp",                     // Mail.ru
+    "ru.yandex.mail",                      // Yandex.Mail
+    "de.tutao.tutanota"                    // Tutanota
+  )
 
   private val sendEmailLauncher =
     registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
@@ -114,15 +165,24 @@ open class ErrorActivity : BaseActivity() {
 
   private suspend fun emailIntent(): Intent {
     val emailBody = buildBody()
-    return Intent(Intent.ACTION_SENDTO).apply {
-      data = Uri.parse("mailto:")
+    return Intent(Intent.ACTION_SEND).apply {
+      type = "text/plain"
       putExtra(Intent.EXTRA_EMAIL, arrayOf(CRASH_AND_FEEDBACK_EMAIL_ADDRESS))
       putExtra(Intent.EXTRA_SUBJECT, subject)
       val file = fileLogger.writeLogFile(
         this@ErrorActivity,
         activityKiwixErrorBinding?.allowLogs?.isChecked == true
       )
-      putExtra(Intent.EXTRA_TEXT, "$emailBody\n\nDevice Logs:\n$${file.readText()}")
+      file.appendText(emailBody)
+      val path =
+        FileProvider.getUriForFile(
+          this@ErrorActivity,
+          applicationContext.packageName + ".fileprovider",
+          file
+        )
+      addFlags(android.content.Intent.FLAG_GRANT_READ_URI_PERMISSION)
+      putExtra(android.content.Intent.EXTRA_STREAM, path)
+      putExtra(Intent.EXTRA_TEXT, emailBody)
     }
   }
 

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/error/ErrorActivity.kt
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/error/ErrorActivity.kt
@@ -140,22 +140,22 @@ open class ErrorActivity : BaseActivity() {
 
   // List of supported email apps
   private val supportedEmailPackages = listOf(
-    "com.google.android.gm",               // Gmail
-    "com.zoho.mail",                       // Zoho Mail
-    "com.microsoft.office.outlook",        // Outlook
-    "com.yahoo.mobile.client.android.mail",// Yahoo Mail
-    "me.bluemail.mail",                    // BlueMail
-    "ch.protonmail.android",               // ProtonMail
-    "com.fsck.k9",                         // K-9 Mail
-    "com.maildroid",                       // Maildroid
-    "org.kman.AquaMail",                   // Aqua Mail
-    "com.edison.android.mail",              // Edison Mail
-    "com.readdle.spark",                   // Spark
-    "com.gmx.mobile.android.mail",          // GMX Mail
-    "com.fastmail",                        // FastMail
-    "ru.mail.mailapp",                     // Mail.ru
-    "ru.yandex.mail",                      // Yandex.Mail
-    "de.tutao.tutanota"                    // Tutanota
+    "com.google.android.gm", // Gmail
+    "com.zoho.mail", // Zoho Mail
+    "com.microsoft.office.outlook", // Outlook
+    "com.yahoo.mobile.client.android.mail", // Yahoo Mail
+    "me.bluemail.mail", // BlueMail
+    "ch.protonmail.android", // ProtonMail
+    "com.fsck.k9", // K-9 Mail
+    "com.maildroid", // Maildroid
+    "org.kman.AquaMail", // Aqua Mail
+    "com.edison.android.mail", // Edison Mail
+    "com.readdle.spark", // Spark
+    "com.gmx.mobile.android.mail", // GMX Mail
+    "com.fastmail", // FastMail
+    "ru.mail.mailapp", // Mail.ru
+    "ru.yandex.mail", // Yandex.Mail
+    "de.tutao.tutanota" // Tutanota
   )
 
   private val sendEmailLauncher =

